scripting/get5.sp

```diff
@@ -1503,22 +1481,11 @@ public void WriteBackup() {
     return;
   }
 
-  char folder[PLATFORM_MAX_PATH];
-  g_RoundBackupPathCvar.GetString(folder, sizeof(folder));
-  ReplaceString(folder, sizeof(folder), "{MATCHID}", g_MatchID);
+  char folder[PLATFORM_MAX_PATH + 1];
+  char variableSubstitutes[][] = {"{MATCHID}"};
+  int variableListSize = sizeof(variableSubstitutes);
 
-  int backupFolderLength = strlen(folder);
-  if (backupFolderLength > 0 &&
-      (folder[0] == '/' || folder[0] == '.' || folder[backupFolderLength - 1] != '/' ||
-       StrContains(folder, "//") != -1)) {
-    LogError(
-        "get5_backup_path must end with a slash and must not start with a slash or dot. It will be reset to an empty string! Current value: %s",
-        folder);
-    folder = "";
-    g_RoundBackupPathCvar.SetString(folder, false, false);
-  } else {
-    CreateBackupFolderStructure(folder);
-  }
+  folder = CheckAndCreateFolderPath(g_RoundBackupPathCvar, variableSubstitutes, variableListSize);
```

Comment:
To follow the standard we normally use for strings, you should probably provide a buffer and buffer size argument to the function instead of returning a `char[]`.